tests/charms/tls_certificates_interface/v1/test_tls_certificates_v1_requires.py

```diff
@@ -0,0 +1,594 @@
+
+        self.harness.charm.certificates.request_certificate_revocation(
+            certificate_signing_request=certificate_signing_request
+        )
+
+        unit_relation_data = self.harness.get_relation_data(
+            relation_id=relation_id, app_or_unit=self.harness.charm.unit
+        )
+
+        self.assertEqual({"certificate_signing_requests": "[]"}, unit_relation_data)
+
+    def test_given_no_csr_in_relation_data_when_request_certificate_revocation_then_nothing_is_done(
+        self,
+    ):
+        remote_app = "tls-certificates-provider"
+        relation_id = self.harness.add_relation(
+            relation_name=self.relation_name, remote_app=remote_app
+        )
+        certificate_signing_request = b"whatever csr"
+
+        self.harness.update_relation_data(
+            relation_id=relation_id, app_or_unit=self.harness.charm.unit.name, key_values={}
+        )
+
+        self.harness.charm.certificates.request_certificate_revocation(
+            certificate_signing_request=certificate_signing_request
+        )
+
+        unit_relation_data = self.harness.get_relation_data(
+            relation_id=relation_id, app_or_unit=self.harness.charm.unit
+        )
+
+        self.assertEqual(dict(), unit_relation_data)
+
+    @patch(f"{LIB_DIR}.TLSCertificatesRequiresV1.request_certificate_creation")
+    @patch(f"{LIB_DIR}.TLSCertificatesRequiresV1.request_certificate_revocation")
+    def test_given_certificate_revocation_success_when_request_certificate_renewal_then_certificate_creation_is_called(  # noqa: E501
+        self, _, patch_certificate_creation
+    ):
+        old_csr = b"whatever old csr"
+        new_csr = b"whatever new csr"
+
+        self.harness.charm.certificates.request_certificate_renewal(
+            old_certificate_signing_request=old_csr, new_certificate_signing_request=new_csr
+        )
+
+        patch_certificate_creation.assert_called_with(certificate_signing_request=new_csr)
+
+    @patch(f"{LIB_DIR}.TLSCertificatesRequiresV1.request_certificate_creation")
+    @patch(f"{LIB_DIR}.TLSCertificatesRequiresV1.request_certificate_revocation")
+    def test_given_certificate_revocation_failed_when_request_certificate_renewal_then_certificate_creation_is_called_anyway(  # noqa: E501
+        self, patch_certificate_revocation, patch_certificate_creation
+    ):
+        old_csr = b"whatever old csr"
+        new_csr = b"whatever new csr"
+        patch_certificate_revocation.side_effect = RuntimeError()
+
+        self.harness.charm.certificates.request_certificate_renewal(
+            old_certificate_signing_request=old_csr, new_certificate_signing_request=new_csr
+        )
+
+        patch_certificate_creation.assert_called_with(certificate_signing_request=new_csr)
+
+    @patch(f"{BASE_CHARM_DIR}._on_certificate_available")
+    def test_given_csr_in_unit_relation_data_and_certificate_in_remote_relation_data_when_relation_changed_then_orchestrator_available_event_emitted(  # noqa: E501
```

Comment:
Is this normal to have an `orchestrator_available_event` in this lib?

Reply:
Oups, no. Fixed